Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.cpp

```diff
@@ -401,11 +432,11 @@ MediaPlayer::SupportsType MediaPlayerPrivateGStreamerMSE::supportsType(const Med
     if (!ok)
         height = 0;
 
-    if (width > MEDIA_MAX_WIDTH || height > MEDIA_MAX_HEIGHT)
+    if (width > mediaMaxWidth || height > mediaMaxHeight)
         return result;
 
     float framerate = parameters.type.parameter("framerate"_s).toFloat(&ok);
-    if (ok && framerate > MEDIA_MAX_FRAMERATE)
+    if (ok && framerate > mediaMaxFrameRate)
```

Comment:
Same as above: mediaMaxFramerate being zero should mean "no limit". Add a condition to apply the comparison only if there's a valid limit.

Reply:
Regarding to maxFrameRate: should it be also applied in case the resolution is lower than specified limit? For example limit is: 1920x1080@30 (maxFrameRate = 30) and supportsType is for lower resolution with higher frameRate (ex.: resolution=720×480, frameRate: 60)?

Reply:
Applying the frame rate limit only if the resolution is the maximum one makes sense, yes.

I don't know how to treat corner cases like if the webpage is asking for 1900x1070@60, or 720x480@200. I guess we can take care of those in the future if they become a problem.